.github/workflows/wheels.yml

```diff
@@ -0,0 +1,125 @@
+# Workflow to build and test wheels
+name: Wheel builder
+
+on:
+  schedule:
+    # Nightly build at 3:42 A.M.
+    - cron: "42 3 */1 * *"
+  push:
+    tags:
+      # Final releases, release candidates and post-releases
+      - "[0-9]+.[0-9]+.[0-9]+(rc[0-9]+)?(.post[0-9]+)?"
```

Comment:
I am wondering if it would be easier to use:

```
on:
  release:
    types: [published]
```

WDYT @ogrisel, @thomasjpfan?

Reply:
I am not familiar enough with github actions to understand what this would entail.

Reply:
Alternatively, instead of relying on a regexp on tags, we could run this for any push to the main repo with the `[cd build]` tag.

Reply:
The current workflow with tags is:

0. Update 0.99.X branch.
1. Create git tag, which uploads to staging.
2. Check that tests past and wheels are uploaded to staging.
3. Create a github release.

With a `[cd build]` commit trigger on master, it enables the workflow:

1. Create PR to `0.99.X` with updates.
3. Merge with `0.99.X` when happy, but remember to include the `[cd build]`. 
4. If we forget to have `[cd build]`, directly push to `0.99.X` with `[cd build]`. (Maybe automate this so we do not forget)
5. Check that checks pass and wheels are uploaded to staging.
6. Create git tag.

The `[cd build]` commit is a little nicer because we can check the `0.99.X` branch before creating a tag.

Given all this, I would prefer to trigger for all pushes to a `0.99.X` branch. These branches are releases branches, so we should have CI test the wheels.

Reply:
I also prefer being able to explicit control when triggering the build with commit message labels on any branches without having to push a tag first.

Third-party systems might monitor our public tags so ideally we don't want to have to delete and re-push the same tag in case of a CI problem on our end to avoid having those third party system trip.